### PR TITLE
Fix lines table overflow detection segfault

### PR DESCRIPTION
**Pull request by Cheaterman**

### CODE REVIEW

**Comment by Zeex** [comment]
Could you write a few lines to explain what this patch does? And where the values `33` and `0x10000` come from?

**Comment by Cheaterman** [comment]
I think the title explains it pretty well. You're currently reading unmapped memory to find where the AMX ends, and (wrongly) expecting zeroes when you do (instead of garbage or an outright crash).

The values have to do with AMX structure. I honestly do not remember. This is tested, working, and most importantly _correct_ (at least more than the original code...).

**Comment by Zeex** [comment]
OK, that's great, thanks

File: src/amx/amxdbg.c [hunk 1]
```diff
@@ -145,21 +145,27 @@ int AMXAPI dbg_LoadInfo(AMX_DBG *amxdbg, FILE *fp)
...
-    #if BYTE_ORDER==BIG_ENDIAN
-      for (index = 0; index <= dbghdr.lines; index++) {
-        amx_AlignCell(&linetbl[index].address);
-        amx_Align32((uint32_t*)&linetbl[index].line);
-        line++;
-      } /* for */
-    #endif
-    ptr += ((uint32_t)dbghdr.lines + 1) * sizeof(AMX_DBG_LINE);
-  } /* while */
+  unsigned int biggest_possible_size = (
```
Comment (Zeex) [inline src/amx/amxdbg.c:149]: The source code of AMX is written in ANSI C, which does not allow declaring variables after statements. Let's stick to that convention to not introduce compilation issues and for consistency sake.

**Review by Zeex — CHANGES_REQUESTED** [review]
@Cheaterman Thanks for submitting this PR. Please check my comments if you still want to have this merged.

**Comment by Zeex** [comment]
Ended up implementing it a little differently in master.

**Pull request closed without merging**